code/modules/projectiles/projectile/bullets.dm

```diff
@@ -198,8 +198,9 @@
 
 /obj/item/projectile/bullet/meteorshot/on_hit(var/atom/target, var/blocked = 0)
 	..()
-	if(istype(target, /atom/movable))
-		var/atom/movable/M = target
+	var/atom/movable/M = target
+	if(M.anchored)return
+	else if(istype(target, /atom/movable))
```

Comment:
You're casting the target to a movable atom and then accessing a var on it before you know if its a movable, which can be bad. I suggest keeping it similar to the format of what it was previously:
```dm
if(istype(target, /atom/movable))
	var/atom/movable/M = target
	if(M.anchored)
		return
	...
```